File: src/main/java/ru/isb/bot/utils/DateUtils.java
package ru.isb.bot.utils;

import lombok.extern.log4j.Log4j2;

import java.text.ParseException;
import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.Date;
import java.util.TimeZone;

@Log4j2
public class DateUtils {

    private static final Integer PAIR_DURATION = 90;

    public static String calcTimeDiscipline(String startTime) {
        SimpleDateFormat df = new SimpleDateFormat("HH:mm"); //df.parse(startTime)
        try {
            Calendar cal = Calendar.getInstance(TimeZone.getDefault());
            cal.setTime(df.parse(startTime));
            Date startDate = cal.getTime();

            cal.add(Calendar.MINUTE, PAIR_DURATION);
            Date endDate = cal.getTime();

            return String.format("%s:%s - %s:%s",
                    startDate.getHours(),
                    String.valueOf(startDate.getMinutes()).length() < 2 ? "0" + startDate.getMinutes() : startDate.getMinutes(),
                    endDate.getHours(),
                    String.valueOf(endDate.getMinutes()).length() < 2 ? "0" + endDate.getMinutes() : endDate.getMinutes());
        } catch (ParseException e) {
            log.error("Invalid format date", e);
            return "с **:** по **:**";
        }
    }
}

File: src/main/java/ru/isb/bot/utils/DateUtils.kt
package ru.isb.bot.utils

import org.apache.logging.log4j.kotlin.Logging
import org.apache.logging.log4j.kotlin.logger
import java.text.ParseException
import java.text.SimpleDateFormat
import java.util.*

class DateUtils : Logging {

    companion object {
        private const val PAIR_DURATION = 90

        fun calcTimeDiscipline(startTime: String?): String {
            val df = SimpleDateFormat("HH:mm") //df.parse(startTime)
            return try {
                val cal = Calendar.getInstance(TimeZone.getDefault())

                cal.setTime(df.parse(startTime))
                val startDate = cal.time

                cal.add(Calendar.MINUTE, PAIR_DURATION)
                val endDate = cal.time
                String.format(
                    "%s:%s - %s:%s",
                    startDate.hours,
                    if (startDate.minutes.toString().length < 2) "0" + startDate.minutes else startDate.minutes,
                    endDate.hours,
                    if (endDate.minutes.toString().length < 2) "0" + endDate.minutes else endDate.minutes
                )
            } catch (e: ParseException) {
                logger.error("Invalid format date", e)
                "с **:** по **:**"
            }
        }
    }

}